Sens and subgroups

alainamstutz opened the pull request:
This branch deals with the control outcomes (positive: diabetes complications / negative: fractures) and subgroup analyses.
Following changes:
1) `data_process.R`: Updated the `cox_stop` argument for `cox-ipw` of the control outcomes. For now, added both, a) counting control outcome events after landmark and b) counting control outcome events only after pandemic. (b) mimics what naturally happens in the primary cox analysis since no covid outcomes can occur before pandemic start.
2) `prep_sg.R`: New R script to prepare the subsets to run `cox-ipw` in each subgroup. Also includes updating the `cox_stop` argument.
3) `project.yaml`: Added the control outcomes as well as all 12 subgroup analyses.

venexia commented on analysis/prep_sg.R:
```diff
@@ -0,0 +1,99 @@
+studyend_date <- as.Date(study_dates$studyend_date, format = "%Y-%m-%d")
+
+
+# Subset the data ---------------------------------------------------------
```
This approach creates and maintains multiple copies of the data in the environment, which is not advisable. I would recommend that you create a function where you pass the subgroup name in as a parameter. The function should do the filtering, prepare and save the dataset, and then clean up. You will then only have the full copy of the data that you are drawing from and the subgroup you are working on in the environment at any one time.

venexia commented on analysis/prep_sg.R:
```diff
@@ -0,0 +1,99 @@
+  filter(cov_cat_ethnicity == "White")
+df_nonwhite <- df %>%
+  filter(cov_cat_ethnicity != "White") # includes unknown
+df_imd1 <- df %>%
```
I wonder if the IMD subgroups would be better powered if implemented as 'IMD<5' versus 'IMD>=5' (assuming you have extracted deciles).

venexia reviewed (changes requested):
Hi @alainamstutz. The addition of the `cox-ipw` actions in the YAML looks correct, as does `data_process`. However, the project does not run all from the command line - it stops with an error at `table1`. If I bypass that and try to run a subgroup (e.g. `cox_sg_nonobese`), it stops with an error for `prep_sg`. I have also suggested some improvements to the structure of the `prep_sg` script.

alainamstutz commented:
Hi @venexia , thanks for the feedback - working on it. However: Very strange that it is not running on your side. In my command line (`opensafely run run_all -f`) and on GitHub this branch is running without error?!

venexia commented:
... It was a me thing. I just got it to run without error. Sorry about that!

alainamstutz commented:
Hi @venexia , 
- `prep_sg.R`: I adapted - without saving all datasets first in the environment. Hope that's now as you had it in mind.
- IMD: I realize I only extracted quintiles. I will add the deciles in a next dataset run, but would run it as it is for now and see re sample size, ok?
- r:latest: I updated all actions to r:v2 and all R script actions run fine - except the ones (actions: table1 and ps) where I use tbl_summary() from gtsummary. There seems to be a versioning issue, I just can't to get it running, esp. accessing the meta_data; I always get a NULL object. Locally it runs fine. I tried for way too long now the last days. I changed it back to r:latest for these two actions and will try again sometime later. Hope that's ok for now?

Thank you very much for your review!
